src/NSwag.AspNetCore/Middlewares/AspNetCoreToOpenApiMiddleware.cs

```diff
@@ -0,0 +1,92 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AspNetCoreToOpenApiMiddleware.cs" company="NSwag">
+//     Copyright (c) Rico Suter. All rights reserved.
+// </copyright>
+// <license>https://github.com/NSwag/NSwag/blob/master/LICENSE.md</license>
+// <author>Rico Suter, mail@rsuter.com</author>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Internal;
+using Microsoft.AspNetCore.Routing.Template;
+using Microsoft.Extensions.ObjectPool;
+
+namespace NSwag.AspNetCore.Middlewares
+{
+    internal class AspNetCoreToOpenApiMiddleware
```

Comment:
This new middleware supports all registered documents in one instance, similar to the existing `SwaggerUi3Settings<T>.SwaggerRoutes` approach. Is there a reason you'd rather not do this?